.github/workflows/unit-tests.yml

```diff
@@ -0,0 +1,82 @@
+name: unit tests
+
+on:
+  push: # run on every push or PR to any branch
+  pull_request:
+  schedule: # run automatically on main branch each Tuesday at 11am
+    - cron: "0 16 * * 2"
+
+env:
+  NODE_VERSION: 10
+
+jobs:
+  js-unit:
+    name: javascript unit tests
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: ${{ env.NODE_VERSION }}
+      - uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+      - run: npm ci
+      - run: npm run test:unit
+      - uses: codecov/codecov-action@v1
+        with:
+          flags: javascript
+
+  python-unit:
+    name: python unit tests
+    runs-on: ubuntu-latest
+    services:
+      db:
+        image: mariadb
+        env:
+          MYSQL_DATABASE: ppa
+          MYSQL_ROOT_PASSWORD: ppa
+        ports:
+          - 3306:3306
+      solr:
+        image: solr:6.6
+        ports:
+          - 8983:8983
+    steps:
+      - uses: actions/checkout@v2
+      - run: docker cp solr_conf ${{ job.services.solr.id }}:/opt/solr/server/solr/configsets/ppa
+      - run: docker exec --user root ${{ job.services.solr.id }} /bin/bash -c "chown -R solr:solr /opt/solr/server/solr/configsets/ppa"
+      - run: echo "PYTHON_VERSION=$(cat .python-version)" >> $GITHUB_ENV
+      - uses: actions/setup-python@v2
+        with:
+          python-version: ${{ env.PYTHON_VERSION }}
+      - uses: actions/cache@v2
+        with:
+          path: ~/.cache/pip
+          key: ${{ runner.os }}-pip-${{ hashFiles('**/requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-pip-
+      - run: pip install -r dev-requirements.txt
+      - run: cp ci/testsettings.py ppa/local_settings.py
+      - run: python -c "import uuid; print('SECRET_KEY = \'%s\'' % uuid.uuid4())" >> ppa/local_settings.py
+      - run: py.test --cov=./ --cov-report=xml
+      - uses: codecov/codecov-action@v1
+        with:
+          flags: python
```

Comment:
oh, cool — hadn't seen these before

Reply:
yeah, I _think_ this is working as intended? I checked codecov and it does show both the typescript and python as part of the same coverage report, so fingers crossed it'll continue to work without issue. we used the same setup on s&co, fwiw